src/app/components/resources/resources.component.ts

```diff
@@ -12,30 +12,44 @@ import {UpdateResourceComponent} from '../edit-resource/update-resource.componen
 })
 export class ResourcesComponent implements OnInit {
   resources: Resource[];
+  private error = false;
 
   constructor(private resourceService: ResourceService, private languageService: LanguageService, private modalService: NgbModal) {}
 
   ngOnInit(): void {
+    this.loadResources();
+  }
+
+  private handleError(error: any): void {
+    console.error(error);
+    this.error = true;
+  }
+
+  private loadResources(): void {
     this.resourceService.getResources('translations,pages').then(resources => {
       this.resources = resources;
       this.resources.forEach(r => (this.loadTranslations(r)));
     });
   }
 
   openCreateModal(): void {
-    this.modalService.open(CreateResourceComponent);
+    const modalRef: NgbModalRef = this.modalService.open(CreateResourceComponent);
+    modalRef.result.then(() => this.loadResources(), error => console.log(error));
   }
 
   openUpdateModal(resource: Resource): void {
-    this.modalService.open(UpdateResourceComponent).componentInstance.resource = resource;
+    const modalRef: NgbModalRef = this.modalService.open(UpdateResourceComponent);
+    modalRef.componentInstance.resource = resource;
+    modalRef.result.then(() => this.loadResources(), error => console.log(error));
   }
 
   loadTranslations(resource): void {
     resource['latest-drafts-translations'].forEach((translation) => {
       this.languageService.getLanguage(translation.language.id, 'custom_pages').then((language) => {
         translation.language = language;
         translation.is_published = translation['is-published'];
-      });
+      })
+        .catch(error => this.handleError(error));
```

Comment:
I think you could just do `.catch(this.handleError);` if you wanted. The anonymous function isn't necessary since you're just passing the same params through.

It's fine how you have it though...